README.md

```diff
@@ -1,273 +1,440 @@
-dependencies {
-    benchmarksCompile sourceSets.main.output + sourceSets.main.runtimeClasspath 
-}
-```
+1. **Create Benchmark Class**: Create a class in your source set where you'd like to add the benchmark. Annotate this class with `@State(Scope.Benchmark)`.
 
-You can also add output and compileClasspath from `sourceSets.test` in the same way if you want 
-to reuse some of the test infrastructure.
+    ```kotlin
+    @State(Scope.Benchmark)
+    class MyBenchmark {
 
+    }
+    ```
 
-Register `benchmarks` source set:
+2. **Set up Parameters and Variables**: Define variables needed for the benchmark.
 
-```groovy
-benchmark {
-    targets {
-        register("benchmarks")    
+    ```kotlin
+    private val size = 10
+
+    private val list = ArrayList()
+    ```
+
+3. **Initialize Resources**: Within the class, you can define any setup or teardown methods using `@Setup` and `@TearDown` annotations respectively. These methods will be executed before and after the entire benchmark run.
+
+    ```kotlin
+    @Setup
+    fun prepare() {
+        for (i in 0 until size) {
+            list.add(i)
+        }
     }
-}
-```
 
-For a Kotlin Multiplatform project:
+    @TearDown
+    fun cleanup() {
+        list.clear()
+    }
+    ```
+
+4. **Define Benchmark Method**: Next, create methods that you would like to be benchmarked within this class and annotate them with `@Benchmark`.
 
-Define a new compilation in whichever target you'd like (e.g. `jvm`, `js`, etc):
-```groovy
-kotlin {
-    jvm {
-        compilations.create('benchmark') { associateWith(compilations.main) }
+    ```kotlin
+    @Benchmark
+    fun benchmarkMethod(): Int {
+        return list.sum()
     }
-}
-```
+    ```
+
+Your final benchmark class will look something like this:
+
+    @State(Scope.Benchmark)
+    class MyBenchmark {
+
+        private val size = 10
+
+        private val list = ArrayList()
+
+        @Setup
+        fun prepare() {
+            for (i in 0 until size) {
+                list.add(i)
+            }
+        }
+
+        @Benchmark
+        fun benchmarkMethod(): Int {
+            return list.sum()
+        }
+
+        @TearDown
+        fun cleanup() {
+            list.clear()
+        }
+    }
+
+Note: Benchmark classes located in the common source set will be run in all platforms, while those located in a platform-specific source set will be run only in the corresponding platform.
+
+See [writing benchmarks](docs/writing-benchmarks.md) for a complete guide for writing benchmarks.
 
-Register it by its source set name (`jvmBenchmark` is the name for the `benchmark` compilation for `jvm` target):
+### Running Benchmarks
 
-```groovy
+To run your benchmarks in all registered platforms, run `benchmark` Gradle task in your project.
+To run only on a specific platform, run `<target-name>Benchmark`, e.g., `jvmBenchmark`.
+
+For more details about the tasks created by the kotlinx-benchmark plugin, refer to [this guide](docs/tasks-overview.md).
```

Comment:
The same issue with the link.

Reply:
The document is added in a followup PR: https://github.com/Kotlin/kotlinx-benchmark/pull/170